lib/plugins/scoreboard.js

```diff
@@ -19,6 +19,10 @@ function inject (bot) {
     }
 
     if (packet.action === 2) {
+      if (!(packet.name in scoreboards)) {
```

Comment:
in operator has unexpected behavior in JavaScript, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in
In short it operates on Object which is not only a dictionary

One way that works is scoreboards[packet.name] !== undefined

Reply:
We know that scoreboards is a dictionary so that shouldn't really be an issue as far as I am concerned?

Reply:
scoreboards is not analogous to a dictionary, it's an Object. Which has special properties inside of it, relating to the prototype. Even if a property inside scoreboards is undefined, `in` will return true because the property is not considered "empty" inside JS.

Eg.
```js
const x = { y: 2 }
x.y = undefined
'y' in x == true
```

Also, this should probably be using `Object.hasOwn(scoreboards, packet.name)` to not match on prototype data inside the object.

Reply:
Makes sense.

Reply:
Fixed with latest commit.